### ISSUE TITLE
Fix the Blog Post title color picker from FSE

### ISSUE THREAD

**Issue by dugyen** [WordPress/gutenberg#40338]
### Description

Unable to change the blog Post title text color and We need this fix to change the blog title text color as preferred by user and should work according to text color that we choose from the color selector.

### Step-by-step reproduction instructions

1.Update to WordPress 6.0 Beta 1 and Activate Gutenberg 13.0 RC 1, TT2 WP Theme.
2.Appearance > Editor (beta)
3.Select Blog Title to change the color using block option.
4.Here you could see three options ( Text, Background and Links)
5.Select Text and try to replace the text color from color picker.
6.Here you cannot able to change the blog post title text color.
7.Indeed other two options ( background and Links) Color works smoothly.

### Screenshots, screen recording, code snippet

https://e1.pcloud.link/publink/show?code=XZKTVzZDJfBExIFiU0TbNiDttMVxRf9UCX0

### Environment info

- WordPress 6.0 Beta 1, Gutenberg 13.0 RC 1 & TT2 Theme
- Google Chrome, Safari and Firefox
- macOS Big Sur

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by ndiego** [comment]
Good catch @dugyen, this is an interesting one, because if you make the Post Title not a link (using the toggle), you can then change the color as expected. We definitely need some improved UI here. 

![post-title-color](https://user-images.githubusercontent.com/4832319/163582541-ad8b99d8-5c10-414b-ae93-994014b62be4.gif)

**Comment by sabernhardt** [comment]
This can happen in both the Post Title and Site Title blocks (when using the link option).

**Comment by dugyen** [comment]
@ndiego I do agree with you and great thanks.
